Core.ApplicationServices/Organizations/OrganizationRightsService.cs

```diff
@@ -75,6 +90,92 @@ public class OrganizationRightsService : IOrganizationRightsService
             return RemoveRight(right);
         }
 
+        public Maybe<OperationError> RemoveUnitRightsByIds(int organizationId, IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var result = CanModifyOrganization(organizationId);
+            if (result.HasValue)
+            {
+                return result;
+            }
+
+            var rightsToDelete = new List<OrganizationUnitRight>();
+            foreach (var rightId in rightIds)
+            {
+                var unitRight = _unitRightRepository.GetByKey(rightId);
+                if (unitRight == null)
+                    return new OperationError($"Unit right with id: {rightId} not found", OperationFailure.NotFound);
+                if (!_authorizationContext.AllowModify(unitRight.Object))
+                    return new OperationError("User is not allowed to perform this operation", OperationFailure.Forbidden);
+                
+                rightsToDelete.Add(unitRight);
+            }
+            var userIds = rightsToDelete.Select(x => x.UserId).ToList();
+
+            _unitRightRepository.RemoveRange(rightsToDelete);
+            foreach (var userId in userIds.Distinct())
+            {
+                _domainEvents.Raise(new AdministrativeAccessRightsChanged(userId));
+            }
+            _unitRightRepository.Save();
+            transaction.Commit();
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> TransferUnitRightsByIds(int organizationId, int targetUnitId, IEnumerable<int> rightIds)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var result = CanModifyOrganization(organizationId);
+            if (result.HasValue)
+            {
+                return result;
+            }
+
+            foreach (var rightId in rightIds)
+            {
+                var unitRight = _unitRightRepository.GetByKey(rightId);
+                if (unitRight == null)
+                {
+                    return new OperationError(OperationFailure.NotFound);
+                }
+
+                if (!_authorizationContext.AllowModify(unitRight.Object))
+                {
+                    return new OperationError(OperationFailure.Forbidden);
+                }
+                
+
+                unitRight.ObjectId = targetUnitId;
+
+                _unitRightRepository.Update(unitRight);
+                _domainEvents.Raise(new AdministrativeAccessRightsChanged(unitRight.UserId));
+            }
+            _unitRightRepository.Save();
+            transaction.Commit();
+            return Maybe<OperationError>.None;
+        }
+
+        private Maybe<OperationError> CanModifyOrganization(int id)
```

Comment:
Can indicates a "boolean" result
The right name for this would be "AuthorizeOrganizationModification"